NUMERIC overflow causes TDS error

bsatarupa opened the pull request:
### Description
NUMERIC overflow causes TDS error for aggregate functions

Before fix, expected resultant precision for aggregate function SUM() was always same as individual precision which caused numeric overflow error even when #digits in sum became 1 more than precision of individual operand.

The fix includes sending default numeric precision(i.e., 38) on wire whenever we encounter aggregate function SUM() and AVG(); and setting resultant scale = max(6, input scale) for aggregate function AVG()

Task: BABEL-3157
Signed-off-by: Satarupa Biswas <satarupb@amazon.com>


### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Commit:
NUMERIC overflow causes TDS error

Fix includes sending default numeric precision(i.e., 38) on wire whenever we encounter aggregate function SUM() and AVG(); and setting resultant scale = max(6, input scale) for aggregate function AVG()

Task: BABEL-3157
Signed-off-by: Satarupa Biswas <satarupb@amazon.com>

surendravishnoi reviewed (dismissed):
Commit message can be more descriptive for both problem and solution part

Commit:
NUMERIC overflow causes TDS error for aggregate functions

Fix includes sending default numeric precision(i.e., 38) on wire whenever we encounter aggregate function SUM() and AVG(); and setting resultant scale = max(6, input scale) for aggregate function AVG()

Task: BABEL-3157
Signed-off-by: Satarupa Biswas satarupb@amazon.com

Commit:
NUMERIC overflow causes TDS error for aggregate functions

Before fix, expected resultant precision for aggregate function SUM() was always same as individual precision which caused numeric overflow error even when #digits in sum became 1 more than precision of individual operand.

The fix includes sending default numeric precision(i.e., 38) on wire whenever we encounter aggregate function SUM() and AVG(); and setting resultant scale = max(6, input scale) for aggregate function AVG()

Task: BABEL-3157
Signed-off-by: Satarupa Biswas satarupb@amazon.com

nasbyj commented on contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c:
```diff
@@ -660,11 +660,22 @@ resolve_numeric_typmod_from_exp(Node *expr)
+			 * function sum(); resultant precision should be tds_default_numeric_precision 
 			 */
 			if (aggFuncName && strlen(aggFuncName) == 3 &&
 					(strncmp(aggFuncName, "sum", 3) == 0))
```
I realize this is pre-existing code, but why aren't we just using `strcmp` here instead of having to invite `strlen` to the party as well?

nasbyj reviewed (approved):
I'm not a fan of the use of strncmp, but not going to block because of that.

hong-yuan reviewed (approved):
:shipit:
